modules/redis-lock/src/index.ts

```diff
@@ -1,57 +1,55 @@
 import { Node } from "@counterfactual/types";
-import { createHandyClient, IHandyRedis } from "handy-redis";
+import Redis from "ioredis";
 import nodeFetch from "node-fetch";
+import Redlock from "redlock";
 import uuid from "uuid";
 
-export class RedisLockService implements Node.ILockInterface {
-  private client?: IHandyRedis;
+export class RedisLockService implements Node.ILockService {
+  private redlock: Redlock;
 
   constructor(redisUrl: string) {
-    this.client = createHandyClient({ url: redisUrl });
+    const redis = new Redis(redisUrl);
+    this.redlock = new Redlock([redis], {
+      // the expected clock drift; for more details
+      // see http://redis.io/topics/distlock
+      driftFactor: 0.01, // time in ms
+
+      // the max number of times Redlock will attempt
+      // to lock a resource before erroring
+      retryCount: 10,
+
+      // the time in ms between attempts
+      retryDelay: 200, // time in ms
+
+      // the max time in ms randomly added to retries
+      // to improve performance under high contention
+      // see https://www.awsarchitectureblog.com/2015/03/backoff.html
+      retryJitter: 200, // time in ms
+    });
   }
 
   async acquireLock(
     lockName: string,
     callback: (...args: any[]) => any,
     timeout: number,
   ): Promise<any> {
-    let retval = null;
-    let rejectReason = null;
-    const unlockKey = await this.getOrCreateLock(lockName, timeout);
-
-    try {
-      retval = await callback();
-    } catch (e) {
-      // TODO: check exception... if the lock failed
-      rejectReason = e;
-    } finally {
-      await this.releaseLock(lockName, unlockKey!);
-    }
+    // acquire lock
+    // if this function errors out, presumably it is because the lock
+    // could not be acquired. this will bubble up to the caller
+    const lock = await this.redlock.lock(lockName, timeout);
 
-    if (rejectReason) throw new Error(rejectReason);
-
-    return retval;
-  }
+    // run callback
+    const retVal = await callback();
```

Comment:
without the try/catch/finally pattern, if `callback` throws, the lock will not be released until whatever its configured max timeout value is.

Reply:
Disregard this comment, try/catch/finally was added back here: 18a638e4e347133982549c762fc8c555c4deb027